pkg/agent/flowexporter/exporter/exporter_test.go

```diff
@@ -0,0 +1,179 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package exporter
+
+import (
+	"strings"
+	"testing"
+	"time"
+	"unicode"
+
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/vmware-tanzu/antrea/pkg/agent/flowexporter"
+	ipfixtest "github.com/vmware-tanzu/antrea/pkg/agent/flowexporter/ipfix/testing"
+	ipfixentities "github.com/vmware/go-ipfix/pkg/entities"
+)
+
+func TestFlowExporter_sendTemplateRecord(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	mockIPFIXExpProc := ipfixtest.NewMockIPFIXExportingProcess(ctrl)
+	mockTempRec := ipfixtest.NewMockIPFIXRecord(ctrl)
+	flowExp := &flowExporter{
+		nil,
+		mockIPFIXExpProc,
+		nil,
+		12,
+		time.Second,
+		256,
+	}
+	// Following consists of all elements that are in IANAInfoElements and AntreaInfoElements (globals)
+	// Need only element name and other are dummys
+	elemList := make([]*ipfixentities.InfoElement, 0)
+	for _, ie := range IANAInfoElements {
+		elemList = append(elemList, ipfixentities.NewInfoElement(ie, 0, 0, 0, 0))
+	}
+	for _, ie := range IANAReverseInfoElements {
+		elemList = append(elemList, ipfixentities.NewInfoElement(ie, 0, 0, 29305, 0))
+	}
+	for _, ie := range AntreaInfoElements {
+		elemList = append(elemList, ipfixentities.NewInfoElement(ie, 0, 0, 0, 0))
+	}
+	// Expect calls for different mock objects
+	tempBytes := uint16(0)
+	var templateRecord ipfixentities.Record
+
+	mockTempRec.EXPECT().PrepareRecord().Return(tempBytes, nil)
+	for i, ie := range IANAInfoElements {
+		mockIPFIXExpProc.EXPECT().GetIANARegistryInfoElement(ie, false).Return(elemList[i], nil)
+		mockTempRec.EXPECT().AddInfoElement(elemList[i], nil).Return(tempBytes, nil)
+	}
+	for i, ie := range IANAReverseInfoElements {
+		split := strings.Split(ie, "_")
+		runeStr := []rune(split[1])
+		runeStr[0] = unicode.ToLower(runeStr[0])
+		mockIPFIXExpProc.EXPECT().GetIANARegistryInfoElement(string(runeStr), true).Return(elemList[i+len(IANAInfoElements)], nil)
+		mockTempRec.EXPECT().AddInfoElement(elemList[i+len(IANAInfoElements)], nil).Return(tempBytes, nil)
+	}
+	for i, ie := range AntreaInfoElements {
+		mockIPFIXExpProc.EXPECT().GetAntreaRegistryInfoElement(ie, false).Return(elemList[i+len(IANAInfoElements)+len(IANAReverseInfoElements)], nil)
+		mockTempRec.EXPECT().AddInfoElement(elemList[i+len(IANAInfoElements)+len(IANAReverseInfoElements)], nil).Return(tempBytes, nil)
+	}
+	mockTempRec.EXPECT().GetRecord().Return(templateRecord)
+	mockTempRec.EXPECT().GetTemplateElements().Return(elemList)
+	// Passing 0 for sentBytes as it is not used anywhere in the test. In reality, this IPFIX message size
+	// for template record of above elements.
```

Comment:
Second sentence does not seem correct grammatically?

Reply:
done

Reply:
fixed.